### PR TITLE
Add friction coefficient parameters

### PR DESCRIPTION
**Pull request by jennuine**
Adds to gazebo_ros_wheel_slip plugin friction coefficient parameters for individual wheels

Requires: https://github.com/osrf/gazebo/pull/3219

### CODE REVIEW

File: gazebo_plugins/src/gazebo_ros_wheel_slip.cpp
Comment (jacobperron) [inline gazebo_plugins/src/gazebo_ros_wheel_slip.cpp]: Maybe it's just because I haven't had my coffee yet, but IMO the code would be more readable if we kept the validation logic separate for slip and friction parameters. I.e. just duplicate this block for friction coefficient parameters.

**Review by jacobperron — COMMENTED** [review]
LGTM, but we'll have to wait for the Gazebo changes to be released before CI will pass here.

File: gazebo_plugins/src/gazebo_ros_wheel_slip.cpp
Comment (jennuine) [reply gazebo_plugins/src/gazebo_ros_wheel_slip.cpp]: I was attempted to avoid code duplication but I agree that your suggestion is more readable 0ef6f4018a455e9164f43658a4b3d84fd06f8fc9

**Review by jacobperron — APPROVED** [review]
LGTM pending CI

**Comment by jennuine** [comment]
@osrf-jenkins run tests please

**Comment by jennuine** [comment]
@ros-pull-request-builder retest this please

**Comment by scpeters** [comment]
we should wrap the new API calls in an `#if` statement based on the available gazebo version, like the following:

* https://github.com/ros-simulation/gazebo_ros_pkgs/blob/388a6d7d68700efa32a3fb3b0ebe9c9240f778d2/gazebo_ros/src/gazebo_ros_init.cpp#L37-L41

**Comment by jennuine** [comment]
> we should wrap the new API calls in an `#if` statement based on the available gazebo version, like the following:
> 
> * https://github.com/ros-simulation/gazebo_ros_pkgs/blob/388a6d7d68700efa32a3fb3b0ebe9c9240f778d2/gazebo_ros/src/gazebo_ros_init.cpp#L37-L41

Done 54bf8b002f770fd202c05ec52890a293ada48469

**Review by jacobperron — APPROVED** [review]

File: gazebo_plugins/test/test_gazebo_ros_wheelslip.cpp
Comment (scpeters) [inline gazebo_plugins/test/test_gazebo_ros_wheelslip.cpp:309]: sorry one last thing; how do you feel about applying this extra patch: https://github.com/ros-simulation/gazebo_ros_pkgs/commit/8a627b06d37edf6d88f387b3f81df2c1f205cc3d

it confirms that we receive the expected number of parameters. If `parameters_received` is empty, this test would currently not complain since it would skip all the assertions

Comment (jennuine) [reply gazebo_plugins/test/test_gazebo_ros_wheelslip.cpp:309]: Done 85084c31a3c702aaeeb5331c4afd494c00bb15ca

**Review by scpeters — APPROVED** [review]

**Pull request merged**
